sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/SupportsVacuum.java

```diff
@@ -0,0 +1,18 @@
+package org.apache.spark.sql.connector.catalog;
+
+import org.apache.spark.annotation.Experimental;
+/**
+ * A mix-in interface for {@link Table} vacuum support. Data sources can implement this
+ * interface to provide the ability to perform table maintenance on request of the user.
+ */
+@Experimental
+public interface SupportsVacuum {
+
+  /**
+   * Performs maintenance on the table.  This often includes removing unneeded data and
+   * deleting stale records.
+   *
+   * @throws IllegalArgumentException If the vacuum is rejected due to required effort.
```

Comment:
Previous comment seems to be go away so adding again (please avoid force pushing unless there's some conflict):

Throwing IllegalArgumentException sounds really weird if there's no argument. IMHO that should be some exception (even a new class) clearly representing the intention.

Reply:
Why allow a source to reject a vacuum operation?

If vacuuming is supported, then the source should always do the operation when requested. I don't see a reason why the source would need to reject it.